WooCommerce/Classes/POS/Presentation/CardPresentPaymentAlert.swift

```diff
@@ -1,191 +1,43 @@
 import SwiftUI
 
 struct CardPresentPaymentAlert: View {
-    // TODO: Figure out whether this is the right choice vs @StateObject
-    // Using @StateObject resulted in the alert not being updated when
-    // alertViewModel changed – possibly we needed to invalidate the view's id.
-    @ObservedObject private var viewModel: CardPresentPaymentAlertSwiftUIViewModel
+    private let viewModel: CardPresentPaymentAlertViewModel
 
-    init(alertViewModel: CardPresentPaymentAlertViewModel) {
-        self.viewModel = .init(alertViewModel: alertViewModel)
+    init(viewModel: CardPresentPaymentAlertViewModel) {
+        self.viewModel = viewModel
     }
 
     var body: some View {
-        BasicCardPresentPaymentAlert(viewModel: viewModel.alertViewModel)
-            .sheet(item: $viewModel.wcSettingsWebViewModel) { webViewModel in
-                WCSettingsWebView(adminUrl: webViewModel.webViewURL, completion: webViewModel.onCompletion)
-            }
-    }
-}
-
-struct BasicCardPresentPaymentAlert: View {
```

Comment:
💭 Since the `BasicCardPresentPaymentAlert` UI is already mostly matching the existing app, perhaps we could have a similar base SwiftUI alert view for cases where we're showing an alert with a different view model that's not `CardPresentPaymentAlertViewModel`.